src/index.js

```diff
@@ -1,35 +1,382 @@
+  threeDotIcon.classList.add('hover-three-dot');
+  threeDotIcon.innerHTML = '&#8942;';
+  todoListItemContainer.appendChild(threeDotIcon);
+
+  // threeDotIcon.addEventListener('click', () => {
+  //   threeDotIcon.parentElement.style.backgroundColor = '#f2f2f2';
+  // });
+
+  document
+    .getElementById('todo-list-element')
+    .appendChild(todoListItemContainer);
+
+  document
+    .getElementById('todo-list-element')
+    .appendChild(document.createElement('hr'));
+
+  console.log(todoListElement.id);
+
+  todoListCheck.addEventListener('click', () => {
+    if (todoListCheck.checked === true) {
+      todoListCheck.nextElementSibling.style.textDecoration = 'line-through';
+      todoListCheck.nextElementSibling.style.opacity = '0.5';
+
+      todoArray[indx].completed = true;
+    } else {
+      todoListCheck.nextElementSibling.style.textDecoration = 'none';
+      todoListCheck.nextElementSibling.style.opacity = '1';
+
+      todoArray[indx].completed = false;
+    }
+    localStorage.setItem('todoList', JSON.stringify(todoArray));
+  });
+
+  const bin = document.createElement('span');
+  bin.classList.add('hover-bin');
+  todoListElement.parentElement.appendChild(bin);
+
+  todoListElement.addEventListener('click', () => {
+    console.log(todoListElement.getAttribute('data-clicked'));
+
+    if (!todoListElement.getAttribute('data-clicked')) {
+      todoListElement.setAttribute('data-clicked', 'true');
+
+      todoListElement.contentEditable = true;
+      todoListElement.classList.remove('ellipsis');
+      todoListElement.focus();
+      todoListElement.parentElement.style.backgroundColor = '#f6d788';
+      todoListElement.style.cursor = 'text';
+      todoListElement.nextElementSibling.style.display = 'none';
+      bin.innerHTML = '&#128465;';
+      bin.style.display = 'inline';
+      bin.onclick = () => {
+        document.querySelector(`#todo-list-container-${indx} + hr`).remove();
+        document.getElementById(`todo-list-container-${indx}`).remove();
+
+        todoArray.splice(indx, 1);
+        todoArray.forEach((todo, ind) => {
+          todo.index = ind;
+        });
+        localStorage.setItem('todoList', JSON.stringify(todoArray));
+      };
+    } else {
+      todoListElement.removeAttribute('data-clicked');
+
+      todoListElement.contentEditable = false;
+      todoListElement.classList.add('ellipsis');
+      todoListElement.parentElement.style.backgroundColor = 'white';
+      todoListElement.style.cursor = 'pointer';
+      todoListElement.nextElementSibling.style.display = 'inline';
+      bin.innerHTML = '';
+      bin.style.display = 'none';
+
+      const indexOfTodoElement = indx;
+
+      console.log(indexOfTodoElement);
+
+      todoArray.forEach((todo) => {
+        if (todo.index === Number(indexOfTodoElement)) {
+          todo.description = todoListElement.innerText;
+        }
+      });
+      localStorage.setItem('todoList', JSON.stringify(todoArray));
+    }
```

Comment:
- [ ]  As the previous reviewer suggested, kindly ensure that by default, new tasks have the property index set to the value of the new array length (i.e. if a 1st task is added to the list, the index of that task should equal to 1, if a 4th task is added to the list, its index property should have a value of 4,...). :+1: 

![req](https://user-images.githubusercontent.com/70483835/236189012-b9f08916-b445-48cc-bc32-761fed931ec2.png)

Reply:
completed.